modules/misc/tmpfiles.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.systemd.user.tmpfiles;
+in {
+  meta.maintainers = [ maintainers.dawidsowa ];
+
+  options.systemd.user.tmpfiles.rules = mkOption {
+    type = types.listOf types.str;
+    default = [ ];
+    example = [ "L /home/user/Documents - - - - /mnt/data/Documents" ];
+    description = ''
+      Rules for creating and cleaning up temporary files
+      automatically. See
+      <citerefentry><refentrytitle>tmpfiles.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+      for the exact format.
+    '';
+  };
+
+  config = mkIf (cfg.rules != [ ]) {
+    xdg = {
+      dataFile."user-tmpfiles.d/home-manager.conf" = {
+        text = ''
+          # This file is created automatically and should not be modified.
+          # Please change the option ‘systemd.user.tmpfiles.rules’ instead.
+          ${concatStringsSep "\n" cfg.rules}
+        '';
+        onChange = "${pkgs.systemd}/bin/systemd-tmpfiles --user --create";
```

Comment:
I've added this line, so that new files will be created when switching to new generation.